### DISCUSSION TITLE
Renovate does not work with Gitea 1.21.1

### DISCUSSION THREAD

**Discussion by calvinbui** [renovatebot/renovate#26020] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.74.1

### If you're self-hosting Renovate, select which platform you are using.

Gitea or Forgejo

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Since upgrading to Gitea 1.21.1, Renovate receives a 404 error.

The issue is also tracked over at https://github.com/go-gitea/gitea/issues/28252

It could be GItea issue, but It's possible the `issues` endpoint has now changed and renovate will require changes to support it.

### Relevant debug logs

<details><summary>Logs</summary>

```

"loggerErrors": [
  {
    "name": "renovate",
    "level": 50,
    "logContext": "<redacted>",
    "repository": "calvinbui/ansible-monorepo",
    "err": {
      "name": "HTTPError",
      "code": "ERR_NON_2XX_3XX_RESPONSE",
      "timings": {
        "start": 1701214916754,
        "socket": 1701214916754,
        "lookup": 1701214916754,
        "connect": 1701214916755,
        "secureConnect": 1701214916756,
        "upload": 1701214916756,
        "response": 1701214916800,
        "end": 1701214916801,
        "phases": {
          "wait": 0,
          "dns": 0,
          "tcp": 1,
          "tls": 1,
          "request": 0,
          "firstByte": 44,
          "download": 1,
          "total": 47
        }
      },
      "message": "Response code 404 (Not Found)",
       "stack": "HTTPError: Response code 404 (Not Found)\n    at Request.<anonymous> (/opt/containerbase/tools/renovate/37.74.1/node_modules/gotdist/source/as-promise/index.js:118:42)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)",
      "options": {
        "headers": {
          "user-agent": "RenovateBot/37.74.1 (https://github.com/renovatebot/renovate)",
          "accept": "application/json",
          "authorization": "***********",
          "accept-encoding": "gzip, deflate, br"
        },
        "url": "https://gitea.<redacted>/api/v1/repos/calvinbui/ansible-monorepo/issues?state=all&type=issues",
        "hostType": "gitea",
        "username": "",
        "password": "",
        "method": "GET",
        "http2": false
      },
      "response": {
        "statusCode": 404,
        "statusMessage": "Not Found",
        "body": {
          "errors": null,
          "message": "The target couldn't be found.",
          "url": "https://gitea.<redacted>/api/swagger"
        },
        "headers": {
          "alt-svc": "h3=\":443\"; ma=2592000",
          "cache-control": "max-age=0, private, must-revalidate, no-transform",
          "content-length": "105",
          "content-type": "application/json;charset=utf-8",
          "date": "Tue, 28 Nov 2023 23:41:56 GMT",
          "strict-transport-security": "max-age=31536000; includeSubDomains; preload",
          "x-content-type-options": "nosniff",
          "x-frame-options": "SAMEORIGIN",
          "connection": "close"
        },
        "httpVersion": "1.1",
        "retryCount": 0
      }
    },
    "msg": "Repository has unknown error"
  }
]

```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
Thanks for reporting. Hopefully there's an answer from gitea maintainers soon about whether it's a regression which will be fixed

**Reply by viceice** [reply]
Strange, i don't see this on my gitea instance 🤔

**Reply by viceice** [reply]
BTW: my token has full access (all scopes) 🙃

**Reply by HonkingGoose** [reply]
The [Renovate docs about Gitea and Forgejo](https://docs.renovatebot.com/modules/platform/gitea/) has a table of required permissions. @calvinbui do your permissions match those listed in the table?

@viceice maybe something changed in the token permissions in the newer version of Gitea?

**Reply by calvinbui** [reply]
yep, the token has the `all` permission and the user is an Administrator

**Reply by calvinbui** [reply]
I've found the root cause, comment here: https://github.com/go-gitea/gitea/issues/28252#issuecomment-1839911672

I can't find it in the config, but is there anyway to disable the Issues Renovate attempts to create?

**Reply by rarkins** [reply]
I don't think it is as simple as that. According to the log in your comment, the failure happens when Renovate tries to fetch the list of issues, which it does to ensure that a particular issue _does not exist_. So it is not Renovate _creating_ an issue, it is Renovate listing issues.

To answer your question, yes it should be possible to disable all issue creation for Renovate. But that won't solve your problem if my understanding is correct.

To solve the problem, it looks like Renovate would need to _never ever_ even query issues. It would be undesirably hacky to modify Renovate that way, so I would only like to do it if it seems a fix from Gitea is unlikely in the near future.

**Reply by viceice** [reply]
I think we can store the `has_issues` setting from repo init and if issues are disabled just return null for search. will implement that asap 😉

**Reply by viceice** [reply]
we're doing the same on github 🙃

**Reply by rarkins** [reply]
Great, that would be a good solution if this only affects repos with issues disabled in that way

**Reply by viceice** [reply, marked as answer]
- #26134

**Reply by ffluegel** [reply]
It still fails for repos with external issue tracker enabled

**Reply by viceice** [reply]
is there another flag we can check?

**Reply by ffluegel** [reply]
Yes, you could check if `external_tracker` exists in the repo init

**Reply by viceice** [reply]
ok, will add another check

**Reply by viceice** [reply]
- https://github.com/renovatebot/renovate/pull/26228
